Objects/listobject.c

```diff
@@ -31,6 +31,50 @@ get_list_freelist(void)
 }
 #endif
 
+#ifdef Py_GIL_DISABLED
+typedef struct {
+    Py_ssize_t allocated;
+    PyObject *ob_item;
```

Comment:
This needs to be a flexible array member like:

```suggestion
    PyObject *ob_item[];
```

Like `dk_indices` in `PyDictKeysObject`  https://github.com/python/cpython/blob/1e68c4b87633b17da1b602b86f5d23bbe106398f/Include/internal/pycore_dict.h#L167